packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -2,20 +2,183 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:meta/meta.dart';
+
 import 'basic_types.dart';
+import 'diagnostics.dart';
 import 'print.dart';
 
 /// Signature for [FlutterError.onError] handler.
 typedef FlutterExceptionHandler = void Function(FlutterErrorDetails details);
 
 /// Signature for [FlutterErrorDetails.informationCollector] callback
-/// and other callbacks that collect information into a string buffer.
-typedef InformationCollector = void Function(StringBuffer information);
+/// and other callbacks that collect information describing an error.
+typedef InformationCollector = Iterable<DiagnosticsNode> Function();
+
+abstract class _ErrorDiagnostic extends DiagnosticsProperty<List<Object>> {
+  /// This constructor provides a reliable hook for a kernel transformer to find
+  /// error messages that need to be rewritten to include object references for
+  /// interactive display of errors.
+  _ErrorDiagnostic(
+    String message, {
+      DiagnosticsTreeStyle style = DiagnosticsTreeStyle.flat,
+      DiagnosticLevel level = DiagnosticLevel.info,
+    }) : assert(message != null),
+         super(
+           null,
+           <Object>[message],
+           showName: false,
+           showSeparator: false,
+           defaultValue: null,
+           style: style,
+           level: level,
+         );
+
+  /// In debug builds, a kernel transformer rewrites calls to the default
+  /// constructors for [ErrorSummary], [ErrorDetails], and [ErrorHint] to use
+  /// this constructor.
+  //
```

Comment:
is it on purpose that the following is just a regular `//` comment and not a dart docs `///` comment? Why?

Reply:
The problem is we don't want this in the official docs and are worried about someone copying and pasting it so it would show up in the officel docs.  This should not be in the public docs because it references privates. There is a discussion of this issue with Ian.